pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -142,9 +139,20 @@ func (gc *GarbageCollector) Run(ctx context.Context, workers int, initialSyncTim
 
 	logger := klog.FromContext(ctx)
 	logger.Info("Starting controller", "controller", "garbagecollector")
-	defer logger.Info("Shutting down controller", "controller", "garbagecollector")
 
-	go gc.dependencyGraphBuilder.Run(ctx)
+	var wg sync.WaitGroup
+	defer func() {
+		logger.Info("Shutting down controller", "controller", "garbagecollector")
+		gc.attemptToDelete.ShutDown()
```

Comment:
As per https://go.dev/tour/flowcontrol/13 goroutine stacked defers are executed in LIFO order. Can we reorder these to be in the same LIFO order they were executed before unless we meant to reorder them? (In which case I would like to understand why we are reordering them)

Reply:
These queues are associated with different workers running in separate threads, the shutdown order doesn't matter.

Reply:
I did align the order in the end, as you wished.